### PR TITLE
dev: Cleanup RPC spec file

### PR DESCRIPTION
**Pull request by smartcontracts**
## Description
Working on cleaning this file up, also adding a test here that makes sure we reject transactions without a chain ID. Removes `OptimismProvider` entirely for the sake of simplicity.

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: packages/tx-ingestion/prettier-config.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "$schema": "http://json.schemastore.org/prettierrc",
```
Comment (smartcontracts) [inline packages/tx-ingestion/prettier-config.json:2]: Really frustrating but somehow this is necessary.

File: packages/x-domain/test/rpc.spec.ts [hunk 2]
```diff
@@ -4,179 +4,169 @@
...
+      }
+    })
   })
+
+  // // There was a bug that causes transactions to be reingested over
+  // // and over again only when a single transaction was in the
+  // // canonical transaction chain. This test catches this by
+  // // querying for the latest block and then waits and then queries
+  // // the latest block again and then asserts that they are the same.
+  // it('should not reingest transactions', async () => {
```
Comment (smartcontracts) [inline packages/x-domain/test/rpc.spec.ts]: Not sure what this test was accomplishing... very unclear from its contents.

File: packages/x-domain/test/rpc.spec.ts [hunk 3]
```diff
@@ -4,179 +4,166 @@
...
-
-    tx.nonce.should.eq(result.nonce)
-    tx.gasLimit.should.eq(result.gasLimit.toNumber())
-    tx.gasPrice.should.eq(result.gasPrice.toNumber())
-    tx.data.should.eq(result.data)
+  let wallet: ethers.Wallet
+  beforeEach(async () => {
+    // Generate a new random wallet before each test. Otherwise we'll run into stateful issues with
+    // nonces and whatnot.
+    wallet = new ethers.Wallet(makeRandomHexString(64), provider)
```
Comment (tynes) [inline packages/x-domain/test/rpc.spec.ts]: `ethers.Wallet.createRandom`

File: packages/x-domain/test/rpc.spec.ts [hunk 4]
```diff
@@ -4,179 +4,166 @@
...
+    it('should correctly process a valid transaction', async () => {
+      const tx = {
+        ...DEFAULT_TRANSACTION,
+        nonce: 0,
+      }
+
+      const result = await wallet.sendTransaction(tx)
+      await result.wait()
+
+      // "from" is calculated client side here, so
```
Comment (tynes) [inline packages/x-domain/test/rpc.spec.ts]: Dead comment

File: packages/x-domain/test/rpc.spec.ts [hunk 5]
```diff
@@ -4,179 +4,166 @@
...
-        data: '0x',
-        value: 0,
-        chainId: parseInt(chainId, 16) + 1,
-      })
-      assert(false)
-    } catch (e) {
-      err = true
-    }
-    assert(err)
+  describe('eth_estimateGas', () => {
```
Comment (tynes) [inline packages/x-domain/test/rpc.spec.ts:160]: We will need to change the way that this works

File: packages/x-domain/helpers/index.ts [hunk 6]
```diff
@@ -1 +1,12 @@
 export * from './setup'
+
+export const makeRandomHexString = (len: number): string => {
```
Comment (tynes) [inline packages/x-domain/helpers/index.ts]: I don't think we need this

**Review by tynes — APPROVED** [review]

**Pull request merged**